File: Python3/201 - 400/287 Find the Duplicate Number.py
# 72ms, 90.99%; 16MB, 7.65%
class Solution:
    def findDuplicate(self, nums: list) -> int:
        tmp = [False] * len(nums)
        for n in nums:
            if tmp[n]: return n
            tmp[n] = True
        

File: 201 - 400/287 Find the Duplicate Number.cc
// 4ms, 99.83%; 10MB, 42.82%
auto _ = []() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    return 0;
}();

class Solution
{
public:
    int findDuplicate(const vector<int> &nums)
    {
        bool *tmp = new bool[nums.size()] {};
        for (auto n : nums) {
            if (tmp[n]) {
                delete[] tmp;
                return n;
            }
            tmp[n] = true;
        }
        return 0;
    }
};
